modules/effects/arrow-effects/src/main/kotlin/arrow/effects/IO.kt

```diff
@@ -90,7 +100,7 @@ sealed class IO<out A> : IOOf<A> {
     IORunLoop.start(this, cb, null)
 
   fun runAsyncCancellable(onCancel: OnCancel = Silent, cb: (Either<Throwable, A>) -> IOOf<Unit>): IO<Disposable> =
-    IO.async { ccb ->
+    IO.async { conn, ccb ->
```

Comment:
It seems like this is the point where my cancellation should be replaced with yours.